openedx_events/event_bus/avro/deserializer.py

```diff
@@ -29,11 +30,20 @@ def _deserialized_avro_record_dict_to_object(data: dict, data_type, deserializer
     """
     param_deserializers = deserializers or {}
     all_deserializers = {**DEFAULT_DESERIALIZERS, **param_deserializers}
+    data_type_origin = get_origin(data_type)
 
     if deserializer := all_deserializers.get(data_type, None):
         return deserializer(data)
     elif data_type in PYTHON_TYPE_TO_AVRO_MAPPING:
         return data
+    elif PYTHON_TYPE_TO_AVRO_MAPPING.get(data_type_origin) == "array":
```

Comment:
@mariajgrimaldi I have only implemented support for list/array as dicts might require more change.

Reply:
@rgraber Makes sense. 
Just in case we want to support other array like python types like set, the current condition would not require any change.

I can still change it to `data_type_origin == list` if you think my statement does not make sense.

Reply:
I still think we should go with` == list` for clarity

Reply:
@rgraber Done.